### PR TITLE
Add helper methods to create request/response records

### PR DESCRIPTION
**Pull request by riking**
I'm doing some archival work where I need all the subresources of a page together in the same WARC, and while most of the work can be done from wpull, some hosts (photobucket) need special handling or are dead (so the response is being pulled from Wayback), so I need to save responses made from Go to the WARC file.

I deprecated WriteRequestMethodAndHeaders because it's blatantly wrong - it always wrote "/" for the path.

Includes test.

### CODE REVIEW

File: cdx.go [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+	} else {
+		u.Host = host + ")"
+	}
+	u.Scheme = "XXX"
+	return strings.TrimPrefix(u.String(), "XXX://")
+}
+
+// The CDXFormat type maps a CDX header character (the key) to an array index.
+// Values should be contiguous.
+type CDXFormat map[byte]int
```
Comment (b5) [inline cdx.go]: hrm, a little work is duplicated in [datatogether/cdxj](https://github.com/datatogether/cdxj). I can see the merit of having rudimentary support for cdx records in this package (as warc cannot import cdxj), but is there any way your use case can import the cdxj package as well?

If not I'm cool to have these here as long as we add a comment mentioning the other package for a "fuller experience" 😄

File: capture.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		c.ReqBodyReadSeeker.Seek(0, io.SeekStart)
+		return c.ReqBodyReadSeeker
+	}
+	if c.ReqBodyBytesBuffer != nil {
+		return c.ReqBodyBytesBuffer
+	}
+	panic("capturehelper: unable to rewind request body")
+}
+
+// Creates a new request/response record pair for the provided HTTP request and
```
Comment (b5) [inline capture.go]: from `golint ./...`:
`capture.go:69:1: comment on exported function NewRequestResponseRecords should be of the form "NewRequestResponseRecords ..."`

File: cdx.go [hunk 3]
```diff
@@ -0,0 +1,270 @@
...
+	"net/http"
+	"net/url"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
```
Comment (b5) [inline cdx.go]: golint will be angry at the following:
`cdx.go:18:2: exported const CDXCanonizedURL should have comment (or a comment on this block) or be unexported`

File: cdx.go [hunk 4]
```diff
@@ -0,0 +1,270 @@
...
+	_ // 'T'
+	CDXUniqness
+	CDXCompressedOffset
+	_ // 'W'
+	CDXCanonizedHrefURL
+	CDXCanonizedSrcURL
+	CDXCanonizedScriptURL // 'Z'
+)
+const (
+	CDXOriginalURL = 'a' + iota
```
Comment (b5) [inline cdx.go]: `cdx.go:46:2: exported const CDXOriginalURL should have comment (or a comment on this block) or be unexported`

File: cdx.go [hunk 5]
```diff
@@ -0,0 +1,270 @@
...
+}
+
+// The CDXFormat type maps a CDX header character (the key) to an array index.
+// Values should be contiguous.
+type CDXFormat map[byte]int
+
+// Writes the CDX fields that can be determined from the record into the target
+// array.  Not all fields are implemented or can be implemented, see source for
+// details.  Fields not written are left at their original values.
+func (r *Record) CDXLine(format CDXFormat, line []string) error {
```
Comment (b5) [inline cdx.go]: cdx.go:151:1: comment on exported method Record.CDXLine should be of the form "CDXLine ..."

**Review by b5 — COMMENTED** [review]
🎉  This is really, really great. Thanks so much @riking! A few golint nitpicks and a question about package scoping, but otherwise I'm very excited to see this merged.

File: cdx.go (hunk 1 above)
Comment (riking) [reply cdx.go]: Hm, my client code only supports old-style CDX right now. I might pull this out and have it in my archiving code instead.

**Review by b5 — APPROVED** [review]
👍 💯 🥇

**Pull request merged**